File: solution/java/week06-legacy-code/exercise2/src/main/java/report/CsvReportGenerator.java
package report;

import java.util.List;

public class CsvReportGenerator implements CanGenerateReport {
    @Override
    public String generateReport(List<ReportData> data) {
        StringBuilder sb = new StringBuilder("Starting CSV Report Generation...\nCSV Header: ID, Value, Description\n");
        data.forEach(d -> sb.append(d.id()).append(",").append(d.value()).append(",").append(d.description()).append("\n"));
        sb.append("CSV Report Generated Successfully.");
        return sb.toString();
    }
}

File: solution/ts/week06-legacy-code/exercise2/src/csvReportGenerator.ts
import { CanGenerateReport } from './canGenerateReport';
import {ReportData} from "./reportData";

export class CsvReportGenerator implements CanGenerateReport {
    generateReport(data: ReportData[]): string {
        let report = "Starting CSV Report Generation...\nCSV Header: ID, Value, Description\n";
        data.forEach(d => {
            report += `${d.id},${d.value},${d.description}\n`;
        });
        report += "CSV Report Generated Successfully.";
        return report;
    }
}